diff --git media/base/mac/avfoundation_glue.mm media/base/mac/avfoundation_glue.mm
index 4bd6d72..9e87575 100644
--- media/base/mac/avfoundation_glue.mm
+++ media/base/mac/avfoundation_glue.mm
@@ -106,7 +106,6 @@ bool LoadAVFoundationInternal() {
 
 void AVFoundationGlue::InitializeAVFoundation() {
   TRACE_EVENT0("video", "AVFoundationGlue::InitializeAVFoundation");
-  CHECK([NSThread isMainThread]);
   if (g_avfoundation_initialization != INITIALIZE_NOT_CALLED)
     return;
   g_avfoundation_initialization = LoadAVFoundationInternal() ?
