diff --git a/content/common/set_process_title.cc b/content/common/set_process_title.cc
index 74651ca6..700bc09 100644
--- content/common/set_process_title.cc
+++ content/common/set_process_title.cc
@@ -42,8 +42,6 @@ void SetProcessTitleFromCommandLine(const char** main_argv) {
   bool have_argv0 = false;
 
 #if defined(OS_LINUX)
-  DCHECK_EQ(base::PlatformThread::CurrentId(), getpid());
-
   if (main_argv)
     setproctitle_init(main_argv);
 
