diff --git build/config/mac/sdk_info.py build/config/mac/sdk_info.py
index 52b252449aa1..f42345e591e2 100644
--- build/config/mac/sdk_info.py
+++ build/config/mac/sdk_info.py
@@ -3,6 +3,7 @@
 # found in the LICENSE file.
 
 import os
+import itertools
 import subprocess
 import sys
 
@@ -12,16 +13,32 @@ import mac_toolchain
 # This script prints information about the build system, the operating
 # system and the iOS or Mac SDK (depending on the platform "iphonesimulator",
 # "iphoneos" or "macosx" generally).
-#
-# In the GYP build, this is done inside GYP itself based on the SDKROOT
-# variable.
+
+def SplitVersion(version):
+  """Splits the Xcode version to 3 values.
+
+  >>> list(SplitVersion('8.2.1.1'))
+  ['8', '2', '1']
+  >>> list(SplitVersion('9.3'))
+  ['9', '3', '0']
+  >>> list(SplitVersion('10.0'))
+  ['10', '0', '0']
+  """
+  version = version.split('.')
+  return itertools.islice(itertools.chain(version, itertools.repeat('0')), 0, 3)
 
 def FormatVersion(version):
-  """Converts Xcode version to a format required for Info.plist."""
-  version = version.replace('.', '')
-  version = version + '0' * (3 - len(version))
-  return version.zfill(4)
+  """Converts Xcode version to a format required for DTXcode in Info.plist
 
+  >>> FormatVersion('8.2.1')
+  '0821'
+  >>> FormatVersion('9.3')
+  '0930'
+  >>> FormatVersion('10.0')
+  '1000'
+  """
+  major, minor, patch = SplitVersion(version)
+  return ('%2s%s%s' % (major, minor, patch)).replace(' ', '0')
 
 def FillXcodeVersion(settings):
   """Fills the Xcode version and build number into |settings|."""
